app/src/Controller/AdviceController.php

```diff
@@ -45,4 +46,33 @@ public function getWorstAdvice(Request $request, GameManager $gameManager, Advis
         $advice = $advisorManager->getWorstAdvice($game);
         return $this->json($advice);
     }
+
+    /**
+     * Return a board view of the advice
+     *
+     * @param  Request     $request
+     * @param  GameManager $gameManager
+     *
+     * @return Response
+     */
+    public function previewAdvice(Request $request, GameManager $gameManager): Response
+    {
+        $id = $request->attributes->getInt('id');
+        $game = $gameManager->getGame($id);
+
+        $xStart = $request->attributes->getInt('xStart');
```

Comment:
I find it a little messy... Maybe you can isolate functions like `getCoordsStartFromRequest` ? (or `list($coordStart, $coordEnd) = getCoordsFromRequest($request)`) ?